### PR TITLE
Initialization script for service/operation like scaffold in Ruby on Rails

### LINKED ISSUE #1163: Initialize script for service/operation like scaffold in Ruby on Rails
Making a script like RoR's scaffold.
https://github.com/toshitanian/moto/tree/feature/scaffold

Implementing new operation of boto3 to moto3 is done by:

adding function, which name s same as boto3's, to models.py and responses.py
adding response template
adding test of service
etc...
We can make it more effective by making a script like RoR's scaffold. It'll generates appropriate functions and templates in library and test directories, and at README if possible.

### PR DESCRIPTION
**Pull request by toshitanian**
By executing `setup_new_function.py` and selecting service and operation,  you'll be able to add auto-generated code and response template to appropriate file.

```
$ python setup_new_function.py
Select service: elbv2
==Current Implementation Status==
[ ] add_tags
[X] create_listener
[X] create_load_balancer
[X] create_rule
[X] create_target_group
[X] delete_listener
[X] delete_load_balancer
[X] delete_rule
[X] delete_target_group
[X] deregister_targets
[ ] describe_account_limits
[X] describe_listeners
[ ] describe_load_balancer_attributes
[X] describe_load_balancers
[X] describe_rules
[ ] describe_ssl_policies
[ ] describe_tags
[ ] describe_target_group_attributes
[X] describe_target_groups
[X] describe_target_health
[ ] modify_listener
[ ] modify_load_balancer_attributes
[X] modify_rule
[ ] modify_target_group
[ ] modify_target_group_attributes
[X] register_targets
[ ] remove_tags
[ ] set_ip_address_type
[X] set_rule_priorities
[ ] set_security_groups
[ ] set_subnets
=================================
Select Operation: set_subnets
        Initializing service    elbv2
        skip creating   moto/elbv2
        skip creating   moto/elbv2/__init__.py
        skip creating   moto/elbv2/exceptions.py
        skip creating   moto/elbv2/models.py
        skip creating   moto/elbv2/responses.py
        skip creating   tests/test_elbv2
        skip creating   tests/test_elbv2/test_server.py
        skip creating   tests/test_elbv2/test_elbv2.py
        inserting code  moto/elbv2/responses.py
        inserting code  moto/elbv2/models.py
```
The inserted codes are...
```
diff --git a/moto/elbv2/models.py b/moto/elbv2/models.py
index 3c6afe7..08afedb 100644
--- a/moto/elbv2/models.py
+++ b/moto/elbv2/models.py
@@ -539,7 +539,10 @@ class ELBv2Backend(BaseBackend):
             modified_rules.append(given_rule)
         return modified_rules

+    def set_subnets(self, load_balancer_arn, subnets):
+        # implement here
+        return availability_zones

 elbv2_backends = {}
 for region in ec2_backends.keys():
-    elbv2_backends[region] = ELBv2Backend(region)
+    elbv2_backends[region] = ELBv2Backend(region)
\ No newline at end of file
diff --git a/moto/elbv2/responses.py b/moto/elbv2/responses.py
index 3e85351..8c336ac 100644
--- a/moto/elbv2/responses.py
+++ b/moto/elbv2/responses.py
@@ -344,6 +344,15 @@ class ELBV2Response(BaseResponse):
         for tag_key, tag_value in zip(tag_keys, tag_values):
             resource.add_tag(tag_key, tag_value)

+    def set_subnets(self):
+        load_balancer_arn = self._get_param("LoadBalancerArn")
+        subnets = self._get_list_prefix("Subnets.member")
+        availability_zones = self.elbv2_backend.set_subnets(
+            load_balancer_arn=load_balancer_arn,
+            subnets=subnets,
+        )
+        template = self.response_template(SET_SUBNETS_TEMPLATE)
+        return template.render(availability_zones=availability_zones)

 ADD_TAGS_TEMPLATE = """<AddTagsResponse xmlns="http://elasticloadbalancing.amazonaws.com/doc/2015-12-01/">
   <AddTagsResult/>
@@ -957,4 +966,20 @@ SET_RULE_PRIORITIES_TEMPLATE = """<SetRulePrioritiesResponse xmlns="http://elast
   <ResponseMetadata>
     <RequestId>4d7a8036-f3a7-11e5-9c02-8fd20490d5a6</RequestId>
   </ResponseMetadata>
-</SetRulePrioritiesResponse>"""
+</SetRulePrioritiesResponse>""
+
+SET_SUBNETS_TEMPLATE = """<SetSubnetsResponse xmlns="http://elasticloadbalancing.amazonaws.com/doc/2015-12-01/">
+  <ResponseMetadata>
+    <RequestId>1549581b-12b7-11e3-895e-1334aEXAMPLE</RequestId>
+  </ResponseMetadata>
+  <SetSubnetsResult>
+    <AvailabilityZones>
+{% for availabilityzone in availabilityzones %}
+      <member>
+        <ZoneName>{{ availabilityzone.zone_name }}</ZoneName>
+        <SubnetId>{{ availabilityzone.subnet_id }}</SubnetId>
+      </member>
+{{ endfor }}
+    </AvailabilityZones>
+  </SetSubnetsResult>
+</SetSubnetsResponse>"""
\ No newline at end of file
```
Then, You can start writing code.

Now supporting `query` and `json` protocol. `rest-json` partially.

closes #1163

### CODE REVIEW

**Comment by JackDanger** [comment]
@toshitanian Some of our users are on Python 2 still, could you get this working on that?

File: setup_new_function.py [hunk 1]
```diff
@@ -0,0 +1,376 @@
...
+
+
+def get_lib_dir(service):
+    return os.path.join('moto', service)
+
+def get_test_dir(service):
+    return os.path.join('tests', 'test_{}'.format(service))
+
+
+def render_teamplte(tmpl_dir, tmpl_filename, context, service, alt_filename=None):
```
Comment (JackDanger) [inline setup_new_function.py]: Minor: typo in function name

File: setup_new_function.py [hunk 2]
```diff
@@ -0,0 +1,376 @@
+#!/usr/bin/env python
```
Comment (JackDanger) [inline setup_new_function.py]: This file could use a comment at the top explaining what it's useful for, how to use it, and what to do if it stops working with future releases of boto, botocore, etc., (e.g. "Please fix this if it breaks under future versions of boto libraries").

I'd also recommend moving this into the ./scripts/ directory and adding an entry to `Makefile` to invoke it. Something like:
```
scaffold:
  pip install -r requirements-dev.txt
  python scripts/setup_new_function.py
```

Maybe it's better named as ./scripts/scaffold.py or ./scripts/generate_scaffold.py? Your call.

**Comment by JackDanger** [comment]
This is really great, @toshitanian! This would have saved me so much time when I implemented ELBv2, I'd really like to use it in the future!

File: setup_new_function.py (hunk 2 above)
Comment (toshitanian) [reply setup_new_function.py]: got it!
I'll do that :)

Comment (toshitanian) [reply setup_new_function.py]: I also move `implementation_coverage.py` to `scripts` directory.

**Comment by toshitanian** [comment]
@JackDanger fixed. pls review again :)

- ensure working on python2
- move script to appropriate directory and added command to Makefile
- add docstring to script
- fix typo
- fix some bugs :)

File: Makefile [hunk 3]
```diff
@@ -29,3 +29,7 @@ tag_github_release:
 	git push origin `python setup.py --version`
 
 publish: upload_pypi_artifact build_dockerhub_image tag_github_release
+
+scaffold:
+	@pip install -r requirements-dev.txt > /dev/null
+	@python scripts/scaffold.py
```
Comment (JackDanger) [inline Makefile]: If you replace this line with `exec python scripts/scaffold.py` then the user input will work correctly on the command line

**Comment by toshitanian** [comment]
cool, thanks!

**Pull request merged**